cpu/sam3x8e/periph/pwm.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2015 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam3x8e
+ * @{
+ *
+ * @file
+ * @brief       CPU specific low-level PWM driver implementation for the SAM3X8E
+ *
+ * @author      Andreas "Paul" Pauli" <andreas.pauli@haw-hamburg.de>
+ *
+ * @}
+ */
+#include <stdint.h>
+#include "board.h"          /* -> cpu.h ??? */
+#include "periph/pwm.h"     /* "driver", interface of pwm.c ONLY */
+
+/*
+ * guard file in case no PWM device is defined, all #define'd in "periph_conf.h"
+ */
+#if PWM_NUMOF && (PWM_0_EN || PWM_1_EN || PWM_2_EN || PWM_3_EN)
+
+#define ERR_INIT_MODE           (-1)
+#define ERR_INIT_BWTH           (-2)
+#define ERR_SET_CHAN            (-1)
```

Comment:
If there are already constants or macros defined, please let me know where to find them. More general, I prefer to write something like:
```
    return ERR_SET_CHAN;
```
instead of 
```
    return -1;
```